lang/csharp/src/apache/test/Reflect/TestUnion.cs

```diff
@@ -149,7 +235,96 @@ public void NullableTest()
             }
         }
 
+        public enum EnumMood
+        {
+            Happy,
+            Sad,
+            Cranky,
+            Meh
+        }
+
         [TestCase]
+        public void NullableEnumTest()
+        {
+            var nullableSchema = @"
+            [
+                ""null"",
+                { ""type"" : ""enum"", ""name"" : ""EnumMood"", ""symbols"" :
+                    [
+                        ""Happy"",
+                        ""Sad"",
+                        ""Cranky"",
+                        ""Meh""
+                    ]
+                }
+            ]
+            ";
+
+            var schema = Schema.Parse(nullableSchema);
+            EnumMood? moodWrite = EnumMood.Meh;
+
+            var writer = new ReflectWriter<EnumMood?>(schema);
+            var reader = new ReflectReader<EnumMood?>(schema, schema);
+
+            using (var stream = new MemoryStream(256))
+            {
+                var encoder = new BinaryEncoder(stream);
+                writer.Write(moodWrite, encoder);
+                writer.Write(null, encoder);
+                stream.Seek(0, SeekOrigin.Begin);
+
+                var decoder = new BinaryDecoder(stream);
+                var moodRead = reader.Read(decoder);
+                var moodNull = reader.Read(decoder);
+                Assert.AreEqual(moodWrite, moodRead);
+                Assert.IsNull(moodNull);
+            }
```

Comment:
So much boilerplate repeated, a shame they don't simplify this across their tests (do not change this!)